tensorboard/plugins/pr_curve/tf_pr_curve_dashboard/tf-step-selector.html

```diff
@@ -0,0 +1,148 @@
+<!--
+@license
+Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+<link rel="import" href="../paper-slider/paper-slider.html">
+<link rel="import" href="../polymer/polymer.html">
+<link rel="import" href="../tf-card-heading/tf-card-heading.html">
+<link rel="import" href="../tf-color-scale/tf-color-scale.html">
+<link rel="import" href="../tf-imports/lodash.html">
+
+<!--
+  Renders a UI for selecting a step.
+-->
+<dom-module id="tf-step-selector">
+  <template>
+    <div id="step-display-container">
+      step <span id="step-text">[[_stepText]]</span>
+    </div>
+    <div id="run-display-container">
+      <div id="run-color-box" style="background:[[color]];"></div>
+      <div id="run-text">
+        [[run]]
+      </div>
+    </div>
+    <paper-slider
+        id="slider"
+        value="{{_stepIndex}}"
+        immediate-value="{{_stepIndex}}"
+        step="1"
+        type="number"
+        min=0
+        max="[[_maxStepIndex]]"
+        ></paper-slider>
+    <style>
+      #slider {
+        margin: -8px 0 0 -15px;
+        width: calc(100% + 31px);
+      }
+      
+      #step-display-container {
+        margin: 10px 15px 0 0;
+      }
+
+      #run-display-container {
+        font-size: 0.7em;
+      }
+
+      #run-text {
+        display: inline-block;
+      }
+
+      #run-color-box {
+        width: 8px;
+        height: 8px;
+        border-radius: 3px;
+        display: inline-block;
+      }
+    </style>
+  </template>
+  <script>
+  "use strict";
+
+  Polymer({
+    is: "tf-step-selector",
+    properties: {
+      // The run this step slider is for.
+      run: String,
+      // The color to use to visualize the run.
+      color: String,
+      // A mapping between run and list of steps to use for the slider for that
+      // run. This component indexes into this object to retrieve the steps.
+      stepsPerRun: Object,
```

Comment:
Can we use `runToSteps` and similar instead? The _per_ formulation suggests homogeneity—i.e., suggests that the set of steps is the same for all runs. The `foo`To`Bar` pattern is used extensively in existing code. (This also affects `runToStepCap`, etc.)